v_5_2_0/files/config/kube-proxy.yaml

```diff
@@ -4,5 +4,4 @@ clientConnection:
 kind: KubeProxyConfiguration
 mode: iptables
 resourceContainer: /kube-proxy
-clusterCIDR: {{.Cluster.Calico.Subnet}}/{{.Cluster.Calico.CIDR}}
```

Comment:
If we're using the AmazonVPC networking, we should omit the ClusterCIDR because pod IPs are real, routable IPs in the VPC, and they are not in a specific CIDR range that allows us to distinguish them from other IPs.  Omitting the ClusterCIDRcauses kube-proxy never to SNAT when proxying clusterIPs, which is the behavior we want for pods.

Reply:
This requires changes and it must be gated.

Reply:
Why are you adding `DisableCalico` here? AFAIK this is used by azure-operator which is basically doing the same thing. Using Azure route tables and using Calico in policy mode.